language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/BallerinaTextDocumentService.java

```diff
@@ -455,25 +455,23 @@ void setClientCapabilities(LSClientCapabilities clientCapabilities) {
     @Override
     public CompletableFuture<WorkspaceEdit> rename(RenameParams params) {
         return CompletableFuture.supplyAsync(() -> {
-            WorkspaceEdit workspaceEdit = new WorkspaceEdit();
-
             try {
-                RenameContext context = ContextBuilder.buildRenameContext(params.getTextDocument().getUri(),
+                Boolean honorsChangeAnnotations = this.clientCapabilities.getTextDocCapabilities()
+                        .getRename().getHonorsChangeAnnotations();
+                RenameContext context = ContextBuilder.buildRenameContext(params,
                         this.workspaceManager,
                         this.serverContext,
-                        params.getPosition());
+                        honorsChangeAnnotations);
```

Comment:
Try passing the clientCapabilities to the rename context. So internally we can do this calculation.

Reply:
+1. That would be ideal